lib/stub_alloc_pages.c

```diff
@@ -42,6 +42,15 @@
 #include <caml/fail.h>
 #include <caml/bigarray.h>
 
+/* Expose the address of the bigarray (in other words, leak the pointer).
+  */
+CAMLprim value
+caml_address(value t)
+{
+  long data = (long)Caml_ba_data_val(t) ;
+  return Val_long(data);
```

Comment:
BTW, we already have a function that does this (supposedly portable, as long as your pointer fits in a signed int64, and your page size is 4096): https://github.com/mirage/io-page/blob/master/lib_test/stub_get_addr.c

I'll try to merge the two.